app/views/questionnaire.py

```diff
@@ -303,20 +356,52 @@ def submit_answers(routing_path, eq_id, form_type):
         if not sent:
             raise SubmissionFailedException()
 
-        get_questionnaire_store(current_user.user_id, current_user.user_ik).delete()
+        store_submitted_responses_and_session_data(answer_store.answers, metadata)
 
-        session_store = get_session_store()
-        session_store.session_data.submitted_time = datetime.utcnow().isoformat()
-        session_store.save()
+        get_questionnaire_store(current_user.user_id, current_user.user_ik).delete()
 
-        return redirect(url_for(
-            'post_submission.get_thank_you',
-            eq_id=eq_id,
-            form_type=form_type))
+        return redirect(url_for('post_submission.get_thank_you', eq_id=eq_id, form_type=form_type))
     else:
         return redirect(invalid_location.url(metadata))
 
 
+def store_submitted_responses_and_session_data(answers, metadata):
+    submitted_time = datetime.utcnow()
+    submitted_responses_enabled, submission_validity = submitted_responses_validity()
+
+    if submitted_responses_enabled:
+        encrypter = StorageEncryption(current_user.user_id, current_user.user_ik)
+        encrypted_data = encrypter.encrypt_data(
+            {
+                'answers': answers,
+                'metadata': metadata
+            }
+        )
+
+        if current_app.eq['submitted_responses'].insert_item(metadata['tx_id'],
+                                                             encrypted_data, submitted_time.strftime('%s'),
+                                                             submission_validity) is False:
+            submission_validity = 0  # failed load so viewing submission is not possible
+
+    session_store = get_session_store()
+    session_data = session_store.session_data
+    session_data.submitted_time = submitted_time.isoformat()
+    session_data.submission_validity = submission_validity
+    session_store.save()
+
+
+def submitted_responses_validity():
+    enabled = False
+    validity = 0
+    view_submitted_responses = g.schema.json.get('view_submitted_responses')
+    if view_submitted_responses:
+        enabled = view_submitted_responses['enabled']
+        if enabled:
+            validity = view_submitted_responses['timeout']
+
+    return enabled, validity
```

Comment:
I think this method should just return whether the schema has view_submitted_responses enabled or not. We can enforce in the schema that timeout should be greater than 0.